### PR TITLE
Adding ACL Service

### PR DESCRIPTION
**Pull request by rosama86**

### CODE REVIEW

**Comment by rosama86** [comment]
please review and let me know if you have any comments

File: lampetia-security-model/src/main/scala/lampetia/security/model/models.scala [hunk 1]
```diff
@@ -279,7 +298,7 @@ trait SecurityModel {
       }
       object parentResource extends Composite[Option[Resource]] {
         val resourceId =
-          property[Option[ResourceId]]("resourceId")
```
Comment (hossamkarim) [inline lampetia-security-model/src/main/scala/lampetia/security/model/models.scala:282]: Why did you remove the option?

Comment (rosama86) [inline lampetia-security-model/src/main/scala/lampetia/security/model/models.scala:282]: since parent resource is optional, resourceId don't need to be optional too. right?

Comment (rosama86) [inline lampetia-security-model/src/main/scala/lampetia/security/model/models.scala:282]: I think the following code can be improved, but I don't know how yet:
protected def insertAcl(acl: Acl): IO[Int] = {
    val aclm = AclModel
    acl.data.parentResource match {
      case None =>
        aclm.insert(aclm.id := acl.id.bind,
          aclm.data.permission := acl.data.permission.bind.cast(Types.bit(32)),
          aclm.data.subject.subjectId := acl.data.subject.subjectId.bind,
          aclm.data.subject.subjectType := acl.data.subject.subjectType.bind,
          aclm.data.resource.resourceId := acl.data.resource.resourceId.bind,
          aclm.data.resource.resourceType := acl.data.resource.resourceType.bind)
      case Some(pr) =>
        aclm.insert(aclm.id := acl.id.bind,
          aclm.data.permission := acl.data.permission.bind.cast(Types.bit(32)),
          aclm.data.subject.subjectId := acl.data.subject.subjectId.bind,
          aclm.data.subject.subjectType := acl.data.subject.subjectType.bind,
          aclm.data.resource.resourceId := acl.data.resource.resourceId.bind,
          aclm.data.resource.resourceType := acl.data.resource.resourceType.bind,
          aclm.data.parentResource.resourceId := acl.data.parentResource.get.resourceId.bind,
          aclm.data.parentResource.resourceType := acl.data.parentResource.get.resourceType.bind
        )
    }
  }

Comment (hossamkarim) [inline lampetia-security-model/src/main/scala/lampetia/security/model/models.scala:282]: OK.
please add a test case to make sure parentResource can be inserted as None.

Comment (hkarim) [inline lampetia-security-model/src/main/scala/lampetia/security/model/models.scala:282]: Did you change the permission type from bit(32) to something else? Why?

Comment (rosama86) [inline lampetia-security-model/src/main/scala/lampetia/security/model/models.scala:282]: already implemented. please check the first test case "Grant ACL"

Comment (hossamkarim) [inline lampetia-security-model/src/main/scala/lampetia/security/model/models.scala:282]: Well, insert accepts a sequence of couples, so you can construct the 
default couples, and then append a sequence of resourceId and 
resourceType or an empty sequence. Something like:

``` scala
val couples =

    Seq(aclm.id := acl.id.bind, ....)
    ++
    acl.data.parentResource match {
       case None => 
         Seq.empty[Couple]
       case Some(pr) => 
         Seq(aclm.data.parentResource.resourceId := pr.resourceId.bind, ...)
    }
```

This will avoid the (.get) in the code as well

Comment (rosama86) [inline lampetia-security-model/src/main/scala/lampetia/security/model/models.scala:282]: I can't see this, permission still has bit(32) type:

models.scala line 310:

val permission =
property[Permission]("permission").set(sql.`type`("bit(32)"))

but I have changed the consumer to :
implicit lazy val consumePermission: Consume[Permission] = consume[String
].fmap(r => Permission(Integer.parseUnsignedInt(r, 2)))

to convert from binary to decimal

On Fri, Aug 7, 2015 at 6:44 PM, Hossam Karim notifications@github.com
wrote:

> In
> lampetia-security-model/src/main/scala/lampetia/security/model/models.scala
> https://github.com/hkarim/lampetia/pull/2#discussion_r36536926:
> 
> > @@ -279,7 +298,7 @@ trait SecurityModel {
> >        }
> >        object parentResource extends Composite[Option[Resource]] {
> >          val resourceId =
> > -          property[Option[ResourceId]]("resourceId")
> 
> Did you change the permission type from bit(32) to something else? Why?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hkarim/lampetia/pull/2/files#r36536926.

## 

Best Regards,
Radwa Osama

Comment (hkarim) [inline lampetia-security-model/src/main/scala/lampetia/security/model/models.scala:282]: OK, may be I've seen it in a previous commit

**Comment by hkarim** [comment]
It seems this pull request has conflicts, could you please check?

**Comment by rosama86** [comment]
yes, last commit on master was merged manually
https://github.com/hkarim/lampetia/commit/2a651cb56e116fa6d90c77b1a6ce07e75b58a0ee

This caused the conflict. I will try to fix it

**Comment by rosama86** [comment]
done, a new pull request was sent

On Fri, Aug 7, 2015 at 6:58 PM, Hossam Karim notifications@github.com
wrote:

> It seems this pull request has conflicts, could you please check?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hkarim/lampetia/pull/2#issuecomment-128765476.

## 

Best Regards,
Radwa Osama
